### DISCUSSION TITLE
Warning when returning disposed object

### DISCUSSION THREAD

**Discussion by matthiaslischka** [dotnet/roslyn#58998] [Ideas]
Returning a disposed object never makes sense, right? 
Or are there any scenarios where it does?

Something like that `using` keyword here is very tricky to spot and dangerous.

```csharp
public Stream GetFile()
{
 	using var fileStream = File.Create(@"c:\tmp\file.txt");
	return fileStream;
}
```

I was just wondering why there is no compiler warning or maybe I just missed it or have to configure it somehow?

BR Matthias

**Comment by Youssef1313** [comment]
I think this might be more suitable as an analyzer. Consider opening an issue in dotnet/runtime repository for an analyzer request.

**Reply by RikkiGibson** [reply]
There are definitely analyzers out there already for checking misuse of disposable objects. I don't recall if they are in the SDK or are done by third parties though.
